src/libs/geolocation.js

```diff
@@ -7,22 +7,39 @@ const geolocationPermissions = {
   DENIED: 'denied',
 };
 
+const getCurrentPositionAsync = function(options) {
+  return new Promise(function(resolve, reject) {
+    navigator.geolocation.getCurrentPosition(resolve, reject, options);
+  });
+};
+
 export default class GeolocationCheck {
-  static async checkPrompt(successCallback = () => {}) {
+  static async checkPrompt(cb = () => {}) {
     if (!window.navigator.permissions) {
       // Some browsers (Safari, etc) do not implement Permissions API
-      return successCallback();
+      return cb(true);
     }
 
-    return window.navigator.permissions.query({ name: 'geolocation' }).then(async p => {
-      if (p.state === geolocationPermissions.PROMPT) {
-        if (window._GEO_QUESTION_ASKED !== true) {
+    return window.navigator.permissions
+      .query({ name: 'geolocation' })
+      .then(async p => {
+        if (p.state === geolocationPermissions.DENIED) {return cb(false);}
+        if (p.state === geolocationPermissions.GRANTED) {return cb(true);}
+        if (p.state === geolocationPermissions.PROMPT) {
+          if (window._GEO_QUESTION_ASKED === true) {return cb(false);}
```

Comment:
I'm not sure I understand this case. Can you explain to which scenario it corresponds?

Reply:
It seems we don't want to display the geolocation modal multiple times if the user has already seen it, while authorization is still in status 'PROMPT'